plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -146,21 +152,23 @@ public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOExce
                 for (DeleteObjectsRequest deleteRequest : deleteRequests) {
                     try {
                         clientReference.client().deleteObjects(deleteRequest);
+                        outstanding.removeAll(partition);
```

Comment:
You're removing "partition" here which is always the last partition, not the partition that is deleted.

Reply:
🤦‍♂ fixed :) sorry about that. Now using the keys from the delete request here (can't use the ones from the response, since it won't contain already missing blob names).

Reply:
@original-brownbear Are you sure that missing blobs won't be reported in the response?
AWS S3 docs (https://docs.aws.amazon.com/en_us/AmazonS3/latest/API/multiobjectdeleteapi.html) say:
```
Note that, if the object specified in the request is not found, Amazon S3 returns the result as deleted.
```
If they're not reported, they won't be reported in `MultiObjectDeleteException` exception as well, and it would mean that exception handling logic is incorrect.

Reply:
See here https://github.com/elastic/elasticsearch/pull/41693/files#diff-e915a7a019920d21471e0f39c435aa16R176. We have the quiet (non-verbose) option activated, so only keys that encounter an error are included. So for exceptions we're good imo.

Reply:
@original-brownbear sorry, I don't get it. If we're using quiet mode and only errors are reported it means we can not fetch the list of successfully deleted objects from `MultiObjectDeleteException` as well?
From `MultiObjectDeleteException.getDeletedObjects` JavaDoc
```
Returns the list of successfully deleted objects from this request. If
     * {@link DeleteObjectsRequest#getQuiet()} is true, only error responses
     * will be returned from s3.
```

Reply:
Ah fair point :D Added some more logic to account for this case now :)

Reply:
@original-brownbear good. Now, why we mess up with outstanding and performing remove/add to it? Why not just to add errors to the failedBlobs set?

Reply:
Basically just to cover running into some other runtime exception in the loop that is then caught in https://github.com/elastic/elasticsearch/pull/41693/files#diff-e915a7a019920d21471e0f39c435aa16R171 (just in case I guess ... not sure how likely/possible it is running into that).

Reply:
@original-brownbear do you mean running into `AmazonClientException`? Ok, I see your point. Can you please add the comment to the method describing the design decision because of quiet mode and generic `AmazonClientException`?

Reply:
Done :)